DataReport/collectBucketMetricsAndUpdateBucketCapacityInfo.js

```diff
@@ -0,0 +1,126 @@
+const async = require('async');
+const { MaxParallelLimit } = require('../utils/constants');
+
+function isSOSCapacityInfoEnabled(bucketInfo) {
+    /**
+     * a valid SOS CapacityInfo Enabled bucketMD looks like
+     * {
+     *     _capabilities: {
+     *         VeeamSOSApi: {
+     *             SystemInfo: {
+     *                 ...
+     *                 ProtocolCapabilities: {
+     *                     CapacityInfo: true,
+     *                     ...
+     *                 },
+     *             ...
+     *             },
+     *             CapacityInfo: {
+     *                 ...
+     *             },
+     *         }
+     *     },
+     * }
+     */
+    return !!(bucketInfo.getCapabilities() // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi.SystemInfo // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi.SystemInfo.ProtocolCapabilities // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi.SystemInfo.ProtocolCapabilities.CapacityInfo === true // should be true
+        && bucketInfo.getCapabilities().VeeamSOSApi.CapacityInfo); // should exist
+}
+
+function isValidBucketStorageMetrics(bucketMetric) {
+    return bucketMetric
+    && bucketMetric.usedCapacity
+        && typeof bucketMetric.usedCapacity.current === 'number'
+        && typeof bucketMetric.usedCapacity.nonCurrent === 'number'
+    && bucketMetric.usedCapacity.current > -1
+    && bucketMetric.usedCapacity.nonCurrent > -1;
+}
+
+function isValidCapacityValue(capacity) {
+    return (typeof capacity === 'number' && capacity >= 0);
+}
+
+function collectBucketMetricsAndUpdateBucketCapacityInfo(mongoClient, log, callback) {
+    let bucketMetrics;
+    return async.waterfall([
+        next => mongoClient.readCountItems(log, (err, doc) => {
+            if (err) {
+                log.error('an error occurred during readCountItems', {
+                    error: { message: err.message },
+                });
+                return next(err);
+            }
+            if (doc && doc.dataMetrics && doc.dataMetrics.bucket) {
+                bucketMetrics = doc.dataMetrics.bucket;
+            }
+            return next();
```

Comment:
Just to be safe, can we return an error if we do not have any dataMetrics? In this case, `bucketMetrics` is still undefined, but we use it line 81 as an array without any check

Reply:
maybe just return without any error, when there are no buckets, there are no bucket metrics, we should just let the job finish